pudb/var_view.py

```diff
@@ -389,142 +611,101 @@ def walk_value(self, parent, label, value, id_path=None, attr_prefix=None):
+        except Exception:
+            ui_log.exception("Failed to look up attributes on {}"
+                                .format(label))
+
+        keys = [key
+                for ki in key_its
+                for key in ki]
+        keys.sort()
+
+        cnt_omitted_private = cnt_omitted_methods = 0
+
+        for key in keys:
+            if iinfo.access_level == "public":
+                if key.startswith("_"):
+                    cnt_omitted_private += 1
+                    continue
+            elif iinfo.access_level == "private":
+                if key.startswith("__") and key.endswith("__"):
+                    cnt_omitted_private += 1
+                    continue
 
             try:
-                key_its.append(dir(value))
+                with warnings.catch_warnings():
+                    warnings.simplefilter("ignore")
+                    attr_value = getattr(value, key)
+                if inspect.isroutine(attr_value) and not iinfo.show_methods:
+                    cnt_omitted_methods += 1
+                    continue
             except Exception:
-                ui_log.exception("Failed to look up attributes")
-
-            keys = [key
-                    for ki in key_its
-                    for key in ki]
-            keys.sort()
+                attr_value = WatchEvalError()
 
-            cnt_omitted_private = cnt_omitted_methods = 0
+            self.walk_value(new_parent_item,
+                    ".%s" % key, attr_value,
+                    "%s.%s" % (id_path, key))
 
-            for key in keys:
-                if iinfo.access_level == "public":
-                    if key.startswith("_"):
-                        cnt_omitted_private += 1
-                        continue
-                elif iinfo.access_level == "private":
-                    if key.startswith("__") and key.endswith("__"):
-                        cnt_omitted_private += 1
-                        continue
-
-                try:
-                    attr_value = getattr(value, key)
-                    if inspect.isroutine(attr_value) and not iinfo.show_methods:
-                        cnt_omitted_methods += 1
-                        continue
-                except Exception:
-                    attr_value = WatchEvalError()
-
-                self.walk_value(new_parent_item,
-                        ".%s" % key, attr_value,
-                        "%s.%s" % (id_path, key))
-
-            if not keys:
-                if cnt_omitted_private:
-                    label = "<omitted private attributes>"
-                elif cnt_omitted_methods:
-                    label = "<omitted methods>"
-                else:
-                    label = "<empty>"
-                self.add_item(new_parent_item, label, None)
+        if not keys:
+            if cnt_omitted_private:
+                label = "<omitted private attributes>"
+            elif cnt_omitted_methods:
+                label = "<omitted methods>"
```

Comment:
Looked into this. We're actually showing neither. This is only hit if `keys` is empty, in which case `cnt_omitted_private` and `cnt_omitted_methods` will never have a chance to be incremented.

I played around with tweaking this so that we do actually show the right thing, but I'm not sure what "the right thing" is in this case. I guess the UX value is explaining to users why they're not seeing anything under an expanded value and what to consider trying, but then what about the very common case of showing some, but not all of the things? We assume the user knows that they're not seeing everything and don't provide this hint. This is now inconsistent: we either give an indication of everything they're missing but only if they're not seeing anything (under an item that looks like maybe they could expand it to see those items, with no indication of how to actually start seeing those items), or trust that they know what they're doing.

I personally vote that we explicitly go the "trust the user to figure it out" route and ditch this bit of code. The "markers" already help point users in the right direction if they suspect they're not seeing everything.